ltiauthenticator/__init__.py

```diff
@@ -182,7 +182,17 @@ class LTIAuthenticateHandler(BaseHandler):
     that URL after authentication. Else, will send them to /home.
     """
 
-    @gen.coroutine
-    def post(self):
-        user = yield self.login_user()
+    async def post(self):
+        current_user = await self.get_current_user()
```

Comment:
FWIW, there is a blocking API to use here:

```suggestion
        current_user = self.current_user
```

which loads the current user from a cache, set during the `prepare()` stage of the handler.